src/main/java/com/alkemy/ong/service/impl/UserServiceImpl.java

```diff
@@ -114,4 +119,23 @@ public UserRegisterResponse upgradeUser(Long id, UserRegisterRequest userJpa) th
         return UserRegisterResponse.mapToResponse(userRepo.save(userBd));
     }
 
+    @Override
+    public UserRegisterResponse getUserRegisterBy(String authorizationHeader) throws UsernameNotFoundException {
+        String email = jwtTokenUtil.getUserEmail(authorizationHeader);
```

Comment:
Estaria bueno cambiar esto por obtener el Authorization del SecurityContextHolder, y a eso pedirle el principal

Reply:
https://stackoverflow.com/questions/31159075/how-to-find-out-the-currently-logged-in-user-in-spring-boot